core/hashfuncs.h

```diff
@@ -75,9 +76,26 @@ static inline uint32_t hash_djb2_one_float(float p_in,uint32_t p_prev=5381) {
 		uint32_t i;
 	} u;
 
-	// handle -0 case
-	if (p_in==0.0f) u.f=0.0f;
-	else u.f=p_in;
+	if (p_in==0.0f)
+		u.f=0.0;
+	else if (std::isnan(p_in))
+		u.f=std::numeric_limits<float>::quiet_NaN();
+	else
+		u.f=p_in;
+
+	return ((p_prev<<5)+p_prev)+u.i;
+}
+
+// Overload for real_t size changes
+static inline uint32_t hash_djb2_one_float(double p_in,uint32_t p_prev=5381) {
+	union {
+		double d;
+		uint64_t i;
+	} u;
+
+	if (p_in==0.0f) u.d=0.0;
```

Comment:
Would be good to use the same style as in the previous method (so not single line ifs but with a proper indentation).